### PR TITLE
Riley vcf header fixes

### PR DESCRIPTION
**Pull request by rimangan**
A couple of programs (lift and vcfFormat) read in VCF data and write transformed VCF data to an output file but discard the header along the way. This is not really optimal in cases where we need the header information, so I have edited these programs to retain the header in the output file. This was really messy for lift, which has its read function tied into the interval interface, so I decided to simply peek into the VCF file to grab the header, close the file, and then open it again for the records channel. Certainly not a clean solution, open to ideas, but this one works.

### CODE REVIEW

File: cmd/lift/lift.go [hunk 1]
```diff
@@ -44,6 +44,13 @@ func lift(chainFile string, inFile string, outFile string, faFile string, unMapp
...
 
+	if vcf.IsVcfFile(inFile) {
+		tmpOpen := fileio.EasyOpen(inFile)
+		header := vcf.ReadHeader(tmpOpen)
+		vcf.NewWriteHeader(out, header)
+		tmpOpen.Close()
+	}
+
 	//second task, read in intervals, find chain, and convert to new interval
 	inChan := interval.GoReadToLiftChan(inFile)
```
[the remark this answers is not in the document]
Comment (craiglowe) [reply cmd/lift/lift.go:55]: Another option is to have GoReadToLiftChan give back all the header text along with a channel where it will send the data.

**Review by craiglowe — COMMENTED** [review]
I gave two possible suggestions to allow us to only read the file once.  Do people have a preference between the two, or other ideas?

File: cmd/lift/lift.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (rimangan) [reply cmd/lift/lift.go:55]: yes these options can start to get messy given that GoReadToLiftChan works on the lift interface so changes would need to be made in several packages. I think the first option between these two is a little cleaner, so I might lean that way unless other people have strong opinions.

[the remark this answers is not in the document]
Comment (rimangan) [reply cmd/lift/lift.go:55]: Also do you have a clear idea of the performance / correctness consequences of opening a file twice? Will this affect our program or is it just bad style?

**Review by cfauci — DISMISSED** [review]

**Review by DanielSnellings — DISMISSED** [review]

**Review by RavenYL — APPROVED** [review]

**Review by cfauci — APPROVED** [review]

**Review by DanielSnellings — APPROVED** [review]

**Pull request merged**